index.js

```diff
@@ -47,6 +54,18 @@ var cacheControlNoTransformRegExp = /(?:^|,)\s*?no-transform\s*?(?:,|$)/
 
 function compression (options) {
   var opts = options || {}
+  var optsBrotli = opts.brotli || {}
+
+  if (hasBrotliSupport) {
+    // set the default level to a reasonable value with balanced speed/ratio
+    if (optsBrotli.params === undefined) {
+      optsBrotli.params = {}
+    }
+
+    if (optsBrotli.params[zlib.constants.BROTLI_PARAM_QUALITY] === undefined) {
+      optsBrotli.params[zlib.constants.BROTLI_PARAM_QUALITY] = 4
```

Comment:
I think this was better when using `object-assign`, now we would be mutating user input if it exists.

I also think `optsBrotli` is only a marginally better change, I'd probably do something more like this (if you want to go down this route):

```js
var brotliOptions = {
  [zlib.constants.BROTLI_PARAM_QUALITY]: opts.brotliQuality,
};
```

I.e. the nesting of `brotli: { quality: 4 }` also seems reasonable, my main concern was that passing through as-is some of those option may actually interfere with the package working (like flush maybe?). I'm not too familiar with this, so trust your judgement on this.

Reply:
I will use Object.assign again to avoid directly mutating the object. Also, I prefer that developers pass the options directly, as it was done

Reply:
You'll still need to `Object.assign` for `params`, currently if provided you'll mutate it input one. `Object.assign` isn't deep, only shallow. It could be worth having a test for it.

Reply:
The easiest way to write something like this with the current shape of how you want it is something like this: 

```
var brotliParams = {}; 
brotliParams[zlib.constants.BROTLI_PARAM_QUALITY] = 4; 
optsBrotli.params = Object.assign(brotliParams, optsBrotli.params);
```

Reply:
Thanks, and that change also reduces some ifs, so it's great.